### PR TITLE
Add CreateFolder method

### PR DESCRIPTION
**Pull request by gavrie**
I was looking for a method to create a new folder inside a given folder, similar to `folder.CreateDatacenter` and `folder.CreateVM`. This commit adds a suitable `folder.CreateFolder` method.

### CODE REVIEW

**Comment by pietern** [comment]
Hi @gavrie! Thanks for your contribution.

Can you please sign and email our CLA? After that we'll be able to merge this commit. Thanks.

See: https://github.com/vmware/govmomi/wiki

**Comment by gavrie** [comment]
Hi Pieter,

I’ve signed and mailed the CLA.

Regards,

-- Gavrie

File: folder.go [hunk 1]
```diff
@@ -73,6 +73,20 @@ func (f Folder) CreateDatacenter(datacenter string) (*Datacenter, error) {
...
 }
 
+func (f Folder) CreateFolder(name string) (*Folder, error) {
+	req := types.CreateFolder{
+		This: f.Reference(),
+		Name: name,
+	}
+
+	res, err := methods.CreateFolder(f.c, &req)
+	if res != nil {
```
Comment (dougm) [inline folder.go]: Hi @gavrie , thanks for the patch and CLA!  Could you change this to:
```go
if err != nil {
   return err
}
```
And at the end of the func: return NewFolder(...), nil
We use this same pattern in the rest of the code, except Folder.CreateDatacenter (which we should fix too).

**Comment by gavrie** [comment]
@dougm, I've updated the patch as you suggested. I initially followed the example of `CreateDatacenter` even though it seemed non-standard to me...

I haven't changed `CreateDatacenter` for now, since the comment in there suggests there is a specific reason for the nonstandard error handling there.

**Comment by pietern** [comment]
@gavrie Great, thank you. The `CreateDatacenter` inconsistency must have fallen through the cracks. Fixing that in a separate PR. Merging your commit. Thanks again :+1:

**Comment by gavrie** [comment]
Sure, glad to help :smile:

**Pull request merged**